### DISCUSSION TITLE
Mountain Duck for Windows ARM

### DISCUSSION THREAD

**Discussion by chrisymarks** [iterate-ch/cyberduck#16132] [Ideas]
Hey, I'm new to  Mountain Duck and, so far, loving it.

I'm using it on Windows ARM, and given all the new Snapdragon Windows Laptops hitting the market and the fact that the official Google Drive Windows app is incompatible, I imagine many more people following suit.  

Mountain Duck works well as is (I'm assuming it's currently working through the Prism compatibility layer?); however, I'm wondering if there are any plans to publish a version that will work specifically with Windows ARM. 

I'm thinking this may be even faster and more efficient?

**Comment by AliveDevil** [comment]
Hi, my last info was that the exe-installer refused to even install.

We do intend to make Cyberduck and Mountain Duck both compatible with Windows 11 ARM64, though there is a lot of tech-debt that needs to be cleaned up, before we can make this even possible.

First Cyberduck needs to be compatible with ARM64, which is in the process of being possible in the future once #16118 is stable.
Curently Cyberduck depends on x64-versions of sunec, sunmscapi and jna-native - of which sunec and sunmscapi are rebuilt for arm64 compatibility in the mentioned PR above. jna has an arm64 build, but we haven't even looked at whether it is compatible with Java 8.

The updater framework used (WinSparkle) isn't available for arm64, thus we need to get away from that as well (possibly by providing MSIX-installation only) and dropping exe/msi installation.

For Mountain Duck all of the above applies, as well as the additional dependency on providing the shell extension, which contains the context menu and overlay icons for Windows Explorer (and compatible applications).

**Comment by chrisymarks** [comment]
Thanks for the detailed update. I appreciate there's work to do, but pleased to hear Windows ARM64 is on your roadmap.

Just to confirm, Mountain Duck installs and runs seamlessly on the latest Windows ARM64 Snapdragon machines running 24H2. I've read that 24H2 has significant improvements for running x64 applications due to Microsoft's new Prism emulation layer. Perhaps that's why Mountain Duck works fine?

The only issue I've noticed is with the handling of large files, where things seem to just 'hang' for a while, before eventually loading. But I've not actually yet experimented to see if it's any slower than on x64. 

Thanks again for the update and for your great software.

**Comment by mdafer** [comment]
Thank you so much for this amazing tool!

I see that #16118  has been closed, any update regarding cyberduck for Win on ARM?

Thanks so much!

**Reply by dkocher** [reply]
This has been superseded by #16157 which is part of 9.2+ and later. Curently work is going on in #17172.
